tsfc/tsfc_to_loopy.py

```diff
@@ -68,30 +70,45 @@ def index_to_iname(self, index):
 
         return iname
 
-    @staticmethod
-    def _is_cse_eligible(node):
+    def _is_cse_eligible(self, node):
+        if not (isinstance(node, g.Literal) and node.array.shape == ()):
+
+            if isinstance(node, (g.FlexiblyIndexed, g.Indexed)):
+                return False
+            else:
+                return True
+        else:
+            return False
+
         return not (
```

Comment:
Does the code ever get down here?

Reply:
No. That;s the left-over code which is not needed. Will remove it.